.github/workflows/BuildAndDeploy.yaml

```diff
@@ -1,5 +1,8 @@
 name: Build and Deploy
-on: [push]
+on:
+  push:
```

Comment:
should only run this workflow on a push for `main` branch and the repo dispatch. So I would add the main branch here for push.